FirebaseUI/Database/Implementation/FirebaseArray.m

```diff
@@ -20,113 +20,142 @@
 
 #import "FirebaseArray.h"
 
+@interface FirebaseArray ()
+
+/**
+ * The delegate object that array changes are surfaced to.
+ */
+@property(strong, nonatomic) NSMutableArray<FIRDataSnapshot *> * snapshots;
+
+/**
+ * A set containing the query observer handles that should be released when
+ * this array is freed.
+ */
+@property(strong, nonatomic) NSMutableSet<NSNumber *> *handles;
+
+@end
+
 @import FirebaseDatabase;
 
 @implementation FirebaseArray
 
-#pragma mark -
-#pragma mark Initializer methods
-
-- (instancetype)initWithRef:(FIRDatabaseReference *)ref {
-  return [self initWithQuery:ref];
-}
+#pragma mark - Initializer methods
 
 - (instancetype)initWithQuery:(FIRDatabaseQuery *)query {
+  NSParameterAssert(query != nil);
   self = [super init];
   if (self) {
-    self.snapshots = [NSMutableArray array];
-    self.query = query;
+    _snapshots = [NSMutableArray array];
```

Comment:
Nit: I prefer assignment to use the self. keyword as then you know the property modifiers are respected (particularly for copy etc). Reads should use the ivar.